qualtran/_infra/composite_bloq.py

```diff
@@ -831,7 +830,9 @@ def from_signature(
         initial_soqs: Dict[str, SoquetT] = {}
         for reg in signature:
             if reg.side & Side.LEFT:
-                initial_soqs[reg.name] = bb.add_register_from_dtype(reg)
+                register = bb.add_register_from_dtype(reg)
+                if register is not None:
+                    initial_soqs[reg.name] = register
```

Comment:
I think we should probably assert register is not None (?). Since the `if` says that it's a LEFT (or THRU) register -- it should definitely be present in initial_soqs